testng-core/src/main/java/org/testng/JarFileUtils.java

```diff
@@ -112,7 +111,11 @@ private void delete(File f) throws IOException {
     if (f.isDirectory()) {
       for (File c : Objects.requireNonNull(f.listFiles())) delete(c);
     }
-    if (!f.delete()) throw new FileNotFoundException("Failed to delete file: " + f);
+
+    /** * Provide better logging for files temp that fail to be cleaned up. */
+    if (!Files.deleteIfExists(f.toPath())) {
+      Utils.log("TestNG", 2, "Failed to delete non-existent temp file: " + f.getPath());
```

Comment:
The behavior is changed and maybe some users prefere the current one.
Please make the behavior change configurable.
By default the exception looks better because we expect the deletion.

Reply:
Removed the logging option, I agree that it should throw an exception instead.